src/content/api/normalmodulefactory-hooks.md

```diff
@@ -0,0 +1,136 @@
+---
+title: NormalModuleFactory Hooks
+group: Plugins
+sort: 13
+contributors:
+  - iguessitsokay
+---
+
+The `NormalModuleFactory` module is used by the `Compiler` to generate modules. Starting with entry points, it resolves each request, parses the content to find further requests, and keeps crawling through files by resolving all and parsing any new files. At last stage, each dependency becomes a Module instance.
+
+The `NormalModuleFactory` class extends `Tapable` and provides the following
+lifecycle hooks. They can be tapped the same way as compiler hooks:
+
+``` js
+NormalModuleFactory.hooks.someHook.tap(/* ... */);
+```
+
+NormaleModuleFactory creates `Parser` and `Generator` instances which can be accessed by HookMaps. Identifier must be passed to tap into these:
+
+``` js
+NormalModuleFactory.hooks.someHook.for('identifier').tap(/* ... */);
+```
+
+As with the `compiler`, `tapAsync` and `tapPromise` may also be available
+depending on the type of hook.
+
+
+### `beforeResolve`
+
+`AsyncSeriesBailHook`
+
+Fired when a new dependency request is encountered. A dependency can be ignored by returning `false`. Otherwise, it should return undefined to proceed.
```

Comment:
```suggestion
Called when a new dependency request is encountered. A dependency can be ignored by returning `false`. Otherwise, it should return `undefined` to proceed.
```